smart_open/webhdfs.py

```diff
@@ -16,7 +16,10 @@
 import logging
 import urllib.parse
 
-import requests
+try:
+    import requests
+except ImportError:
+    MISSING_DEPS = True
```

Comment:
What are the implications of `MISSING_DEPS`?

Reply:
We import modules like smart_open.http and smart_open.hdfs dynamically. Trapping the ImportError and setting MISSING_DEPS allows us to import these modules fully (for discovering their functionality) but warn the user if they try to actually invoke the functionality with the missing dependencies.

The code that does all that is here: https://github.com/RaRe-Technologies/smart_open/blob/develop/smart_open/transport.py